### ISSUE TITLE
Bot "hold position" crash, MAYBE WolfHud?

### ISSUE THREAD

**Issue by Malidictus** [Kamikaze94/WolfHUD#50]
I'm not sure if this crash is related to WolfHud (the crash log isn't specific to it), but I'm told that the "hold position" mechanic should work properly and it doesn't. A friend of mine has Payday 2 install with nothing on it but BLT and WolfHud. When he attempts to order the bots to hold position, he crashes to the desktop every time.

That's when he's hosting, however. Something on my system overwrites that mechanic entirely so my character just calls the bots and my friend ordering them to hold position as a client in my lobby does nothing. His character yells and nothing happens - nobody crashes. It's when trying to halt bots as his own host that he crashes.

Crash log as follows:

`Application has crashed: C++ exception
[string "lib/managers/hudmanagerpd2.lua"]:1823: attempt to index local 'name_text' (a nil value)

SCRIPT STACK

-------------------------------

Callstack:

         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     ???                                                 
         payday2_win32_release  (???)     zip_get_name                                        
                         ntdll  (???)     RtlFreeHeap                                         
                         ntdll  (???)     ZwQueryVirtualMemory                                

-------------------------------

Current thread: Main

-------------------------------
`

As I said - generic game hud crash, but my suspicion is it may have something to do with the game trying to draw the shield next to the bot's name and tripping up WolfHud in some way. I'm using the Keepers mod and that works just fine with WolfHud, including the shield icon, but it seems like Overkill's version works differently.

**Comment by Kamikaze94** [comment]
Yes, I can confirm its caused by CustomHUD.
Its the small hand icon that gets put after a bots name, that cannot get created due to the different structure of CustomHUD.
Just pushed a little update, to resolve this.
The HUD will show the Hand Icon in the place of Bots down or in custody indicators.
(Will not show on the label above the bots head anymore, as overkills code looks kinda messy at that point (or at least what my small decompiler puts out there...))

Anyways, update pushed for now, should fix the crashing and lets you see if a bot is in hold animation or following.

**Comment by Malidictus** [comment]
After disabling both Keepers and the Movable Intimidated Cop which were blocking the entire mechanic on my side, I found that it does indeed work now. No crashes, bots hold position and they get an arm next to their display in the UI. Question, however - is there any way you can put that symbol in their floating names? I realise it's bad form to go "but X mod does it" but the Keepers (which does basically the same thing) puts a shield icon in the bot's floating name.

Having an icon in the UI is more useful than nothing, but it's extremely hard to use efficiently. I'd need to find which bot has a hand icon, read the name, then find that name in the field. That's as opposed to glancing at a bot and seeing an icon. It's not an urgent thing, but it's something I'd really like to see eventually.

Point is, though - the mod no longer crashes :)

**Comment by Kamikaze94** [comment]
Yeah, I probably will, but it was 7am and I wanted to sleep :P
So I just stopped the crashing, and after that added some kind of indicator to make that feature useable.

_Edit:_
Added the Hand Symbol to the label now.
Also removed it from the HUD for now, it disappears, once the bot gets tazed or downed.
(Hand Icon in label stays, and Bot itself stays too....^^)

**Comment by Malidictus** [comment]
Oh, of course :) I didn't mean to rush you, I was mostly just thinking out loud. Nevertheless, you work pretty dang quickly, I have to say. Thank you kindly both for the fix and the extension.

I didn't get an auto-update for WolfHUD, so I presume we now go back to working on the development build which won't be undergoing constant auto-updates until a major version is finished unless a new crash bug is discovered? Should I just snag the GitHub version to test out this latest set of changes?

**Comment by Kamikaze94** [comment]
Yeah, currently I'm trying to get back to the normal schedule of adding features, having them around a bit, so bugs get (hopefully) discovered, and releasing after some time.
The official update and the following crashing just forced me to soemthing immediently.

And if you want to help find those bugs and aren't too feared of the potential to crash at some point, because I did something stupid, sure, stay on the dev build :P

As the topic of this issue is resolved, I'll close this now.
Thanks for reporting that bug :D

**Comment by Malidictus** [comment]
Follow-up on this issue that I just ran into:

The bot hold position "hand" icon in the UI will persist if a client drops in, into the spot that bot previously held. The player will then have the hand icon permanently over his health in the team display. This does fix itself if another icon is displayed, however. The one I tested with was Swan Song, as the player in question got downed once. As soon as Swan Song expired, the hand went away.

It's not a huge problem but it obscures health and armour and is very clearly not working as intended.
